tests/test_butlerFits.py

```diff
@@ -0,0 +1,214 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import unittest
+import tempfile
+import shutil
+
+import lsst.utils.tests
+
+import lsst.afw.image
+from lsst.afw.image import LOCAL
+from lsst.geom import Box2I, Point2I, Extent2I
+
+from lsst.daf.butler import Config
+from lsst.daf.butler import StorageClassFactory
+from lsst.daf.butler import DatasetType
+from lsst.daf.butler.tests import DatasetTestHelper, makeTestRepo, addDatasetType, makeTestCollection
+
+from lsst.obs.base.exposureAssembler import ExposureAssembler
+
+TESTDIR = os.path.dirname(__file__)
+
+butlerConfig = """
+storageClasses:
+  ExposureCompositeF:
+    inheritsFrom: ExposureF
+datastore:
+  # Want to check disassembly so can't use InMemory
+  cls: lsst.daf.butler.datastores.posixDatastore.PosixDatastore
+  formatters:
+    ExposureCompositeF: lsst.obs.base.fitsExposureFormatter.FitsExposureFormatter
+  composites:
+    disassembled:
+      ExposureCompositeF: True
+"""
+
+# Components present in the test file
+COMPONENTS = {"wcs", "image", "mask", "coaddInputs", "psf", "visitInfo", "variance", "metadata", "photoCalib"}
+
+
+class ButlerFitsTests(DatasetTestHelper, lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a new butler once only."""
+
+        cls.storageClassFactory = StorageClassFactory()
+
+        cls.root = tempfile.mkdtemp(dir=TESTDIR)
+
+        dataIds = {
+            "instrument": ["DummyCam"],
+            "physical_filter": ["d-r"],
+            "visit": [42],
+        }
+
+        cls.creatorButler = makeTestRepo(cls.root, dataIds, config=Config.fromYaml(butlerConfig))
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.root is not None and os.path.exists(cls.root):
+            shutil.rmtree(cls.root, ignore_errors=True)
```

Comment:
This is like this because it's like that all over daf_butler. I hadn't thought about it before.

Reply:
`rmtree` doesn't like `None` but can handle lack of existence. I'm not sure whether tearDownClass is called if setUpClass does not complete.

Reply:
It does not get called. That's one of the reasons I prefer to use `addCleanup` instead of `tearDown`. (Unfortunately, `addClassCleanup` is not available in Python 3.7.)